src/Parsers/CSVParser.php

```diff
@@ -41,10 +42,49 @@ public function from(String $source): StreamParserInterface
 	public function each(callable $function)
 	{
 		$this->start();
-		while($this->read()){
-			$function($this->getCurrentLineAsCollection());
+		try {
+			while($this->read()){
+				if($function($this->getCurrentLineAsCollection()) === false) {
+					break;
+				}
+			}
+		} catch (StopParseException $e) {
```

Comment:
Let's find other way to stop the process.

Reply:
Are you talking about the `$function() === false` or `StopParseException` or both?

Reply:
Maybe we could pass the Parser instance to the callback as the second function, then pause() and resume() methods on the interface and parsers and boolean so the process gets paused or resumed. What do you think?